### ISSUE TITLE
Add example to ElementTree documentation for default namespace (to avoid spurious `ns0:` namespace prefix)

### ISSUE THREAD

**Issue by dechamps** [python/cpython#113425]
### Bug description:

# Minimal reproducer

```python
import xml.etree.ElementTree as ET

print(
    ET.tostring(ET.fromstring("<foo xmlns='somens'>a<bar /></foo>"), encoding="unicode")
)
```

# Expected output

```xml
<foo xmlns="somens">a<bar /></foo>
```

# Actual output

```xml
<ns0:foo xmlns:ns0="somens">a<ns0:bar /></ns0:foo>
```

# Discussion

It would appear that, if a namespace isn't registered, ElementTree will use a `ns0:` namespace prefix by default during serialization, instead of simply using the default namespace.

Now to be fair, technically this is a valid thing to do, as the resulting XML is semantically the same. However I still find this behavior problematic because:

1. It is confusing and violates the principle of least surprise. Users won't expect to see a random namespace prefix just popping out of nowhere.
2. It unnecessarily increases the size of the resulting XML document.
3. Most importantly, the use of namespace prefixes can cause problems with downstream consumers.
   - For example, the SVG viewer in VS Code and in the GitHub file browser get confused if an SVG file uses namespace prefixes. (This is how I stumbled upon this issue)
   - This means ElementTree can turn perfectly working documents into ones that can cause compatibility headaches.

ElementTree should just use the default namespace, and only resort to prefixes if the default namespace is already in use.

# Workaround

In the above example, pass `default_namespace="somens"` to `ET.tostring()`. Note this requires the user to know the namespace URL in advance. If they don't know it in advance they can get it from the root element but ElementTree doesn't make that easy: it requires ugly string parsing on the element name.

### CPython versions tested on:

3.11

### Operating systems tested on:

Linux

### Linked PRs
* gh-147976

**Comment by gaogaotiantian** [comment]
First of all, I think this should be a feature request, instead of a bug, because it is not WRONG and it does not contradict the documentation.

From a feature request point of view, I like the propsed result, but there are a couple of things to concern.

1. This is a BREAKING change. Yes, semantically it's still the same, but like you said, semantics are not the only thing to worry about. This could break someone's code. We need an expert to evaluate the benefit vs the cost. This module has behaved like this for a long time.
2. It's not trivial (from my understanding) because the data we need is lost during parsing (not serializing). As far as I can tell, `<ns0:foo xmlns:ns0="somens">a<ns0:bar /></ns0:foo>` and `<foo xmlns="somens">a<bar /></foo>` are parsed to the same `ElementTree`. So in order to conform to the principle of least surprise, `ET.tostring(ET.fromstring("<ns0:foo xmlns:ns0="somens">a<ns0:bar /></ns0:foo>"), encoding="unicode")` should be `<ns0:foo xmlns:ns0="somens">a<ns0:bar /></ns0:foo>` which is not possible without changing the parsing part.

I'm not saying it's impossible (well very few things are really impossible in software world), but we need further discussion (presumably with some core devs involved) in this. I won't be too optimistic about this request because of the reasons I mentioned above. You might be recommended to use some 3rd party libraries to do this kind of work.

Let's see if this can raise anyone's interests to the topic :)

**Comment by hexagonrecursion** [comment]
```
For example, the SVG viewer in VS Code and in the GitHub file browser get confused if an SVG file uses namespace prefixes. (This is how I stumbled upon this issue)
```

Based on the very limited information you provided this sounds like you should report this as a bug on *their* end - they are the ones who failed to correctly implement the [Namespaces in XML specification](https://www.w3.org/TR/xml-names/#dt-NSDecl).

**Comment by dechamps** [comment]
@hexagonrecursion I agree that it's a bug on their end, but consider the [Robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) - "be conservative in what you send, be liberal in what you accept". Randomly adding namespace prefixes where none are required is not being "conservative in what you send". Fixing ElementTree to produce XML that is less likely to cause problems, *and* fixing bugs in consumers, are not mutually exclusive propositions.

**Comment by merwok** [comment]
This comes from the private _namespaces function, called from the ElementTree.write method (called by tostring). It returns namespaces and qnames, taking into account the default_parameter argument; when it’s none, a namespace prefix is synthesized (`ns0`, `ns1`, etc) so that all namespaces are identified.

So, if you want a serialization that uses a default namespace, use the `default_namespace` parameter.

**Comment by emelinea-hina** [comment]
I've hit this today while updating a 7 years old script, and it was very confusing (and frustrating). I was trying to read an ATOM file my script had written, to update it. I think at least some documentation update on how to avoid this would be very welcome. I have ended up adding random bits of code found on stack overflow without really understanding what they do. In comparison writing XSLT 1.0 sheets is a delight.
